### PR TITLE
Cleanup Enum tests

### PR DESCRIPTION
**Pull request by hughbe**
- Take advantage of xunit
- Add missing tests

Contributes to #6268
Cherry picked from #5490

### CODE REVIEW

File: src/System.Runtime/tests/System/Enum.cs [hunk 1]
```diff
@@ -1,78 +1,82 @@
...
-    [InlineData("Red", false, true, SimpleEnum.Red)]
-    [InlineData(" Red", false, true, SimpleEnum.Red)]
-    [InlineData("Red ", false, true, SimpleEnum.Red)]
-    [InlineData(" red ", true, true, SimpleEnum.Red)]
-    [InlineData(" Red , Blue ", false, true, SimpleEnum.Red | SimpleEnum.Blue)]
-    [InlineData("Blue,Red,Green", false, true, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green)]
-    [InlineData("Blue,Red,Red,Red,Green", false, true, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green)]
-    [InlineData("Red,Blue,   Green", false, true, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green)]
-    [InlineData("B", false, true, SimpleEnum.B)]
-    [InlineData("B,B", false, true, SimpleEnum.B)]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Enum.cs]: Why did you remove this?  It was explicitly being used to test 1-char-long enum names.

Comment (hughbe) [inline src/System.Runtime/tests/System/Enum.cs]: I truly have no idea. Fixed

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Debug Build and Test
Test  Innerloop CentOS7.1 Release Build and Test

File: src/System.Runtime/tests/System/Enum.cs [hunk 2]
```diff
@@ -1,78 +1,82 @@
...
-
-    [InlineData("1", false, true, SimpleEnum.Red)]
-    [InlineData(" 1 ", false, true, SimpleEnum.Red)]
-    [InlineData("2", false, true, SimpleEnum.Blue)]
-    [InlineData("99", false, true, (SimpleEnum)99)]
-    [InlineData("-42", false, true, (SimpleEnum)(-42))]
-    [InlineData("   -42", false, true, (SimpleEnum)(-42))]
-    [InlineData("   -42 ", false, true, (SimpleEnum)(-42))]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Enum.cs]: Why did you remove this case?

File: src/System.Runtime/tests/System/Enum.cs [hunk 3]
```diff
@@ -170,103 +166,84 @@ public static void TestIsDefinedInvalid()
...
     }
 
     [Fact]
-    public static void TestCompareToInvalid()
+    public static void TestCompareTo_Invalid()
     {
-        Assert.Throws<ArgumentException>(null, () => SimpleEnum.Red.CompareTo((sbyte)1)); //Target is an enum type
-        Assert.Throws<ArgumentException>(null, () => SimpleEnum.Red.CompareTo(Int32Enum.One)); //Target is a different enum type
+        Assert.Throws<ArgumentException>(() => SimpleEnum.Red.CompareTo((sbyte)1)); // Target is not an enum type
+        Assert.Throws<ArgumentException>(() => SimpleEnum.Red.CompareTo(Int32Enum.One)); //Target is a different enum type
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Enum.cs]: Why remove the null being passed as the parameter name?

File: src/System.Runtime/tests/System/Enum.cs [hunk 4]
```diff
@@ -466,14 +451,21 @@ public static void TestGetNamesAndValues(Type t, string[] expectedNames, object[
...
-        Assert.Equal(expected, s);
+        if (format.ToUpperInvariant() == "G")
+        {
+            Assert.Equal(expected, e.ToString());
+            Assert.Equal(expected, e.ToString(""));
+            Assert.Equal(expected, e.ToString(null));
+        }
+        // Format string is non-case-sensitive
+        Assert.Equal(expected, e.ToString(format.ToUpperInvariant()));
+        Assert.Equal(expected, e.ToString(format.ToLowerInvariant()));
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Enum.cs]: What about the original case of just:
```Assert.Equal(expected, e.ToString(format));```
?

File: src/System.Runtime/tests/System/Enum.cs [hunk 5]
```diff
@@ -513,7 +529,7 @@ private enum SimpleEnum
         Green = 3,
         Green_a = 3,
         Green_b = 3,
-        B = 4
+        B
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Enum.cs]: Why did you remove the = 4?

**Comment by stephentoub** [comment]
Some questions/comments to be addressed, otherwise LGTM.

**Comment by hughbe** [comment]
Addressed your comments

**Comment by stephentoub** [comment]
LGTM. Thanks.
